fixed-price-subscriptions/client/react/src/Account.js

```diff
@@ -1,196 +1,65 @@
+  return (
+    <section>
+      <hr />
+      <h4>
+        <a href={`https://dashboard.stripe.com/test/subscriptions/${subscription.id}`}>
+          {subscription.id}
+        </a>
+      </h4>
 
-function Account({ location }) {
-  console.log(location);
-  const [accountInformation] = useState(location.state.accountInformation);
-  let [customerPaymentMethod, setCustomerPaymentmethod] = useState(null);
-  let [showChangePriceForm, setShowChangePriceForm] = useState(false);
-  let [subscriptionCancelled, setSubscriptionCancelled] = useState(false);
-  let [newProductSelected, setNewProdctSelected] = useState('');
-  let [selectedProducted, setSelectedProduct] = useState(
-    accountInformation.priceId
-  );
+      <p>
+        Status: {subscription.status}
+      </p>
 
-  useEffect(() => {
-    async function fetchData() {
-      // You can await here
-      const response = await fetch('/retrieve-customer-payment-method', {
-        method: 'post',
-        headers: {
-          'Content-type': 'application/json',
-        },
-        body: JSON.stringify({
-          paymentMethodId: accountInformation.paymentMethodId,
-        }),
-      });
-      const responseBody = await response.json();
-      const paymentMethod =
-        responseBody.card.brand + ' •••• ' + responseBody.card.last4;
+      <p>
+        Card last4: {subscription.default_payment_method.card.last4}
+      </p>
 
-      setCustomerPaymentmethod(paymentMethod);
-    }
-    fetchData();
-  }, [accountInformation.paymentMethodId]);
+      <p>
+        Current period end: {(new Date(subscription.current_period_end * 1000).toString())}
+      </p>
 
-  function handleChangePriceForm() {
-    setShowChangePriceForm(true);
-  }
+      {/* <Link to={{pathname: '/change-plan', state: {subscription: subscription.id }}}>Change plan</Link><br /> */}
+      <Link to={{pathname: '/cancel', state: {subscription: subscription.id }}}>Cancel</Link>
+    </section>
+  )
+}
 
-  function handleClick(key) {
-    setNewProdctSelected(products[key].name);
-  }
+const Account = () => {
+  let [subscriptions, setSubscriptions] = useState([]);
 
-  function cancelSubscription() {
-    console.log(accountInformation.subscription);
-    fetch('/cancel-subscription', {
-      method: 'post',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({
-        subscriptionId: accountInformation.subscription.id,
-      }),
-    })
-      .then((response) => {
-        return response.json();
-      })
-      .then((cancelSubscriptionResponse) => {
-        setSubscriptionCancelled(true);
-      });
-  }
+  useEffect(() => {
+    const fetchData = async () => {
+      const {subscriptions} = await fetch('/subscriptions').then(r => r.json());
 
-  function resetDemo() {
-    localStorage.clear();
-    window.location.href = '/';
+      setSubscriptions(subscriptions.data);
+    }
+    fetchData();
+  }, [subscriptions]);
```

Comment:
Is the idea that we wan to re-run `fetch(/subscriptions` every time the subscriptions array changes? If not, that's what will likely happen here, which might be undesirable since the thing that changes the subscriptions array is the effect itself.

Reply:
Just need to fetch when the page loads.